laterpay/utils.py

```diff
@@ -31,14 +32,21 @@ def signed_query(secret,
                    to ``signing.sign()``. "GET" is a default.
     :param add_timestamp: ``bool`` (default True) - Should a timestamp param be
                           added to the query.
+    :param permalink: ``bool`` (default False) - Generate a permanent
+                (non-expiring) link. If ``True``, ``add_timestamp`` will be
+                ignored and a timestamp will not be added to the query.
     :param signature_param_name: Name of the appended signature param
                                  (default "hmac")
 
     :return: url-encoded and signed query string
     """
     params = signing.normalise_param_structure(params)
 
-    if "ts" not in params and add_timestamp:
```

Comment:
I'd like to have an additional `TypeError` raised when `add_timestamp` is true and `permalink` is true.

Do you have thoughts about renaming the parameter here from `permalink` to `make_permalink`? This would be a bit more explicit about what the parameter is doing. Probably also in https://github.com/laterpay/laterpay-client-python/pull/116/files#diff-0f20a0385651ab80da384355d8ad2f25R272

Reply:
Given that `permalink=True` implies `add_timestamp=False`, I'd like someone using the API not to have to explicitly specify both -- e.g., `lpc.get_buy_url(i, permalink=True)`. Is there a reason why we would want to force a merchant to be explicit with `add_timestamp=False` if they want a permalink?

I'm not strongly for or against `make_permalink`, but I think `permalink` is probably a lot more intuitive from reading the API docs, given that `permalink` is the argument that gets appended. Of course, I am updating the API docs to include the laterpay python client usage, but I think it will be easier to remember than `make_permalink`.

Reply:
I was hoping to safeguard against `lpc.get_buy_url(i, add_timestamp=True, permalink=True)`, where `add_timestamp` and `permalink` are both set explicitly. That's a bit more overhead and probably not worth the effort for now.

I'm fine with sticking with `permalink` over `make_permalink`. The latter feels more in line with `add_timestamp` though. It's a parameter defining an action. `add_permalink` would work as well. I simply feels more in line with the rest of the API around it.

Reply:
I believe that `add_timestamp` isn't a parameter you can pass to `get_buy_url` (https://github.com/laterpay/laterpay-client-python/blob/develop/laterpay/__init__.py#L261). None of the `kwargs` get passed to `signed_url`. So the good news is, we already can't even ask for an invalid link, even if we wanted one. :grin: 

I don't think `add_permalink` is quite right, exactly. In this case, `permalink` isn't something the URL has, it's something it *is*. I'll update it to `is_permalink`. Would that be a good compromise?